tests/ui/type-alias-impl-trait/not_a_defining_use.rs

```diff
@@ -23,8 +22,7 @@ impl Bar for u32 {
 
 fn four<T: Debug, U: Bar>(t: T) -> Two<T, U> {
     (t, <U as Bar>::FOO)
-    //~^ ERROR `U: Bar` is not satisfied
-    //~| ERROR `T` doesn't implement `Debug`
+    //~^ ERROR concrete type differs
```

Comment:
what is this test actually testing :thinking: the error is correct but the name indicates that this wasn't the original intention here